testing/src/cypress/e2e/04-api-institution/02-instition-create-update-location.cy.ts

```diff
@@ -0,0 +1,103 @@
+import InstitutionHelperActions from "../../custom-command/institution/common-helper-functions.cy";
+import Authorization from "../../custom-command/common/authorization";
+import ApiData from "../data/endpoints/api-endpoints.json";
+
+const institutionHelperActions = new InstitutionHelperActions();
+
+function createOrUpdateInstitutionLocation(
+  token: string,
+  id: string,
+  canadian: boolean,
+  create: boolean
+) {
+  const uniqueId = id;
+  let body: object;
+  let method: string;
+  let url: string;
+  let status: number;
+  if (create) {
+    method = "POST";
+    url = `${ApiData.testEnv.baseUrl}${ApiData.endPoints.createInstitutionLocation}`;
+    status = 201;
+  } else {
+    method = "PATCH";
+    url = `${ApiData.testEnv.baseUrl}${ApiData.endPoints.createInstitutionLocation}/1`;
+    status = 200;
+  }
+  if (canadian) {
+    body = {
+      locationName: `Auto-${uniqueId}-location`,
+      institutionCode: "AUTO",
+      addressLine1: `Auto-${uniqueId}-Street`,
+      addressLine2: "14235",
+      selectedCountry: "Canada",
+      country: "Canada",
+      city: `Auto-${uniqueId}-City`,
+      canadaPostalCode: "A1A2A3",
+      primaryContactFirstName: `Auto-${uniqueId}-FirstName`,
+      primaryContactLastName: `Auto-${uniqueId}-LastName`,
+      primaryContactEmail: `Auto@${uniqueId}.com`,
+      primaryContactPhone: "32165496872",
+      submit: true,
+      provinceState: "NS",
+      postalCode: "A1A2A3",
+    };
+  } else {
+    body = {
+      locationName: `Auto-${uniqueId}-location`,
+      institutionCode: "AUTO",
+      addressLine1: `Auto-${uniqueId}-Street`,
+      addressLine2: "14235",
+      selectedCountry: "other",
+      country: `Auto-${uniqueId}-Country`,
+      city: `Auto-${uniqueId}-City`,
+      postalCode: "98654",
+      primaryContactFirstName: `Auto-${uniqueId}-FirstName`,
+      primaryContactLastName: `Auto-${uniqueId}-LastName`,
+      primaryContactEmail: `Auto@${uniqueId}.com`,
+      primaryContactPhone: "32165496872",
+      submit: true,
+      otherCountry: `testAuto-${uniqueId}-Country`,
+      otherPostalCode: "98654",
+    };
+  }
+  cy.request({
+    method: method,
+    url: url,
+    followRedirect: false,
+    headers: {
+      Authorization: `Bearer ${token}`,
+    },
+    body: body,
+  }).then((response) => {
+    expect(response.status).to.be.equal(status);
+  });
+}
+describe("[Institution Create/Update] Verify location create/update", () => {
+  let token: any;
+  const uniqueId1 = institutionHelperActions.getUniqueId();
+  const uniqueId2 = institutionHelperActions.getUniqueId();
```

Comment:
what are these for my understanding

Reply:
we wanted to create a unique location name created every time we run the test. Just for the sake of the tracking